packages/eslint-plugin-template/src/rules/no-duplicate-attributes.ts

```diff
@@ -0,0 +1,89 @@
+import { TmplAstElement } from '@angular/compiler';
+import {
+  createESLintRule,
+  getTemplateParserServices,
+} from '../utils/create-eslint-rule';
+import {
+  BoundAttribute,
+  BoundEvent,
+  TextAttribute,
+} from '@angular/compiler/src/render3/r3_ast';
+
+type Options = [];
+export type MessageIds = 'noDuplicateAttributes';
+export const RULE_NAME = 'no-duplicate-attributes';
+
+export default createESLintRule<Options, MessageIds>({
+  name: RULE_NAME,
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description:
+        'Ensures that no components have duplicate input properties or output event listeners',
+      category: 'Best Practices',
+      recommended: false,
+    },
+    schema: [],
+    messages: {
+      noDuplicateAttributes:
+        'The attribute "{{attributeName}}" was declared multiple times',
+    },
+  },
+  defaultOptions: [],
+  create(context) {
+    const parserServices = getTemplateParserServices(context);
+
+    return {
+      Element({ inputs, outputs, attributes }: TmplAstElement) {
+        const duplicateInputsAndAttributes = findDuplicates([
+          ...inputs,
+          ...attributes,
+        ]);
+
+        const nonTwoWayBindingOutputs = outputs.filter((output) => {
+          return !inputs.find(
+            (input) =>
+              input.sourceSpan.start === output.sourceSpan.start &&
+              input.sourceSpan.end === output.sourceSpan.end,
+          );
+        });
+
+        const duplicateOutputs = findDuplicates(nonTwoWayBindingOutputs);
+
+        const allDuplicates = [
+          ...duplicateInputsAndAttributes,
+          ...duplicateOutputs,
+        ];
+
+        allDuplicates.forEach((duplicate) => {
+          const loc = parserServices.convertNodeSourceSpanToLoc(
+            duplicate.sourceSpan,
+          );
+
+          context.report({
+            messageId: 'noDuplicateAttributes',
+            loc,
+            data: {
+              attributeName: duplicate.name,
+            },
+          });
+        });
+      },
+    };
+  },
+});
+
+function findDuplicates(elements: Array<BoundEvent>): Array<BoundEvent>;
+function findDuplicates(
+  elements: Array<BoundAttribute | TextAttribute>,
+): Array<BoundAttribute | TextAttribute>;
+function findDuplicates(
+  elements: Array<{ name: string }>,
+): Array<{ name: string }> {
+  return elements.filter((element) => {
+    return elements.some(
+      (otherElement) =>
+        otherElement !== element && otherElement.name === element.name,
+    );
+  });
+}
```

Comment:
If we agree that we should only report the duplications (excluding the first occurrence), I have this suggestion that could be faster than two implicit loops (but that's just a suggestion):

```suggestion
type Attribute = TmplAstBoundAttribute | TmplAstBoundEvent | TmplAstTextAttribute;

function findDuplicates(elements: readonly Attribute[]): readonly Attribute[] {
  return elements.reduce<readonly Attribute[]>(
    (previousValue, currentValue, index) =>
      previousValue.concat(
        currentValue.name === elements[index + 1]?.name
          ? elements[index + 1]
          : [],
      ),
    [],
  );
}
```